src/node/structural_hash.cc

```diff
@@ -395,12 +396,73 @@ struct ArrayNodeTrait {
   }
 
   static bool SEqualReduce(const ArrayNode* lhs, const ArrayNode* rhs, SEqualReducer equal) {
+    if (equal.IsPathTracingEnabled()) {
```

Comment:
probably off the topic, but ideally we should find a better place for defining those traits >,<